pom.xml

```diff
@@ -2476,7 +2476,11 @@
                 <configuration>
                     <properties>
                         <configurationParameters>junit.jupiter.execution.timeout.thread.mode.default = SEPARATE_THREAD
-                            junit.jupiter.extensions.autodetection.enabled = true</configurationParameters>
+                            junit.jupiter.extensions.autodetection.enabled = true
+                            junit.jupiter.execution.parallel.enabled = true
+                            junit.jupiter.execution.parallel.mode.default = concurrent
```

Comment:
note that on classes with lifecycle `PER_CLASS` an explicit `@Execution(CONCURRENT)` annotation needs to be added otherwise they run single threaded.

The better solution is to convert `PER_CLASS` tests to use `@BeforeAll` to initialize stuff.

Reply:
Additionally it's a good idea to explicitly specify `junit.jupiter.execution.parallel.mode.classes.default` to `same_thread` otherwise it defaults to value of `junit.jupiter.execution.parallel.mode.default`.

With current values all classes and methods run concurrently - will lead to high resource utilization for most connector tests.

Reply:
> note that on classes with lifecycle PER_CLASS an explicit @Execution(CONCURRENT) annotation needs to be added otherwise they run single threaded.

Yes, I’m aware. I’m introducing changes one by one to see what may cause problems.

> The better solution is to convert PER_CLASS tests to use @BeforeAll to initialize stuff where possible.

How does that solve it? The methods in the class would still run single-threaded in that case.

Reply:
> Additionally it's a good idea to explicitly specify junit.jupiter.execution.parallel.mode.classes.default to same_thread otherwise it defaults to value of junit.jupiter.execution.parallel.mode.default.

A better option is to add explicit mutual exclusion for those tests that cannot run together. We should be able to parallelize cheaper tests even if they are spread out across multiple classes.

See https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-parallel-execution-synchronization